Spigot-API-Patches/0295-Add-PlayerPickItemEvent.patch

```diff
@@ -0,0 +1,112 @@
+From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
+From: Rodney_Mc_Kay <herr-fant@web.de>
+Date: Fri, 7 May 2021 21:46:54 +0200
+Subject: [PATCH] Add PlayerPickItemEvent
+
+
+diff --git a/src/main/java/io/papermc/paper/event/player/PlayerPickItemEvent.java b/src/main/java/io/papermc/paper/event/player/PlayerPickItemEvent.java
+new file mode 100644
+index 0000000000000000000000000000000000000000..b1d71bc4b66a34b998d9784e9719f8cfc7d5a3e3
+--- /dev/null
++++ b/src/main/java/io/papermc/paper/event/player/PlayerPickItemEvent.java
+@@ -0,0 +1,100 @@
++package io.papermc.paper.event.player;
++
++import org.apache.commons.lang.Validate;
++import org.bukkit.entity.Player;
++import org.bukkit.event.Cancellable;
++import org.bukkit.event.HandlerList;
++import org.bukkit.event.player.PlayerEvent;
++import org.bukkit.inventory.ItemStack;
++import org.jetbrains.annotations.NotNull;
++import org.jetbrains.annotations.Nullable;
++
++/**
++ * Event that is fired when a player uses the pick item functionality (middle clicking a block to get the appropriate
++ * item). However, note that this event will only trigger if an item has to be moved from the inventory to the hotbar.
++ * After the handling of this event the contents of the source and the target slot will be swapped and the currently
++ * selected hotbar slot of the player will be set to the target slot.
++ */
++public class PlayerPickItemEvent extends PlayerEvent implements Cancellable {
++    private static final HandlerList handlers = new HandlerList();
++    @NotNull
++    public static HandlerList getHandlerList() {
++        return handlers;
++    }
++
++    private int targetSlot;
++    private int sourceSlot;
++
++    private boolean cancelled;
++
++    public PlayerPickItemEvent(@NotNull Player player, int targetSlot, int sourceSlot) {
++        super(player);
++        this.targetSlot = targetSlot;
++        this.sourceSlot = sourceSlot;
++    }
++
++    /**
++     * Returns the slot the item that is being picked goes into.
++     *
++     * @return hotbar slot (0-8 inclusive)
++     */
++    public int getTargetSlot() {
++        return this.targetSlot;
++    }
++
++    /**
++     * Changes the slot the item that is being picked goes into. Note: This may be equal to the source slot.
++     *
++     * @param targetSlot hotbar slot (0-8 inclusive)
++     */
++    public void setTargetSlot(int targetSlot) {
++        Validate.isTrue(targetSlot >= 0 && targetSlot <= 9, "Target slot must be in range 0 - 9 (inclusive)");
```

Comment:
As you already mentioned prior, this should be in the range of 0 to 8 with both bounds included in the rage.
This is also stated in the javadocs for this method, yet you are validating whether or not the target slot is in the inclusive range of 0 to 9 (while the hotbar only matches slots 0 through 8).

This should be fixed in both the validation call itself as well as in the resulting error message.